EIPS/eip-4519.md

```diff
@@ -0,0 +1,237 @@
+    function updateTimestamp() external; 
+    
+    /// @notice This function lets obtain the tokenID from an address. 
+    /// @dev Everybody can call this function. The code executed only reads from the blockchain.
+    /// @param _addressSA is the address to obtain the tokenID from it.
+    /// @return The tokenID of the token bound to the asset that generates _addressSA.
+    function tokenFromBCA(address _addressSA) external view returns (uint256);
+    
+    /// @notice This function lets know the owner of the token from the address of the asset bound to the token.
+    /// @dev Everybody can call this function. The code executed only reads from the blockchain.
+    /// @param _addressSA is the address to obtain the owner from it.
+    /// @return the owner of the token bound to the asset that generates _addressSA.
+    function ownerOfFromBCA(address _addressSA) external view returns (address);
+    
+    /// @notice This function lets know the user of the token from its tokenID.
+    /// @dev Everybody can call this function. The code executed only reads from the blockchain.
+    /// @param _tokenId is the tokenID of the SmartNFT bound to the asset.
+    /// @return The user of the token from its _tokenId.
+    function userOf(uint256 _tokenId) external view returns (address);
+    
+    /// @notice This function lets know the user of the token from the address of the asset bound to the token.
+    /// @dev Everybody can call this function. The code executed only reads from the blockchain.
+    /// @param _addressSA is the address to obtain the user from it.
+    /// @return The user of the token bound to the asset that generates _addressSA.
+    function userOfFromBCA(address _addressSA) external view returns (address);
+    
+    /// @notice This function lets know how many tokens are assigned to a user.
+    /// @dev Everybody can call this function. The code executed only reads from the blockchain.
+    /// @param _addressUser is the address of the user.
+    /// @return the number of tokens assigned to a user.
+    function userBalanceOf(address _addressUser) external view returns (uint256);
+    
+    /// @notice This function lets know how many tokens of an particular owner are assigned to a user.
+    /// @dev Everybody can call this function. The code executed only reads from the blockchain.
+    /// @param _addressUser is the address of the user.
+    /// @param _addressOwner is the address of the owner.
+    /// @return the number of tokens assigned to a user from an owner.
+    function userBalanceOfAnOwner(address _addressUser, address _addressOwner) external view returns (uint256);
+}
+```
+This interface is an extension of the ERC-721, is compatible with the standard, and needs the ERC-721 interface. ERC-721 interface is available in https://eips.ethereum.org/EIPS/eip-721. Then, metadata and enumeration extensions are compatible and included in this draft.
```

Comment:
sine EIP-165 is required, I would recommend mentioning that `supportsInterface` should be overriden to include the `interfaceId` of this EIP